src/DataAccess/DoctrineTypes/Iban.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\PaymentContext\DataAccess\DoctrineTypes;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\Type;
+use WMDE\Fundraising\PaymentContext\Domain\Model\Iban as WMDEIban;
+
+class Iban extends Type {
+
+	public function getSQLDeclaration( array $column, AbstractPlatform $platform ): string {
+		return 'VARCHAR(255)';
```

Comment:
VARCHAR(255) is probably over the top, VARCHAR(50) should be enough

Reply:
My reasoning here is that because this field is a VARCHAR it will only use as much disk space as the length of the item stored regardless of the specified length. Then Doctrine automagically makes string fields VARCHAR(255) when specifying them in the xml mappings so this makes the Iban column consistent with those.

We can chat about it in the daily anyway.

Reply:
From a domain perspective I'd be in favor of limiting the number of chars (both for IBAN and BIC, also for the payment reference code in Sofort and bank transfer) for two reasons:

- to protect against accidental garbage data insertion (as a last line of defense; of course the data inside the entity should already be length-validated)
- To communicate to developers who inspect the database schema what kind of data we might want

But this is not a blocker for this PR, we can have another chat about it and do a followup commit that touches all DB code if needed.

Reply:
I'm going to merge this so I can work on the [bank data validation](https://phabricator.wikimedia.org/T300489).  We can go through the fields later and set lengths.